File: 217Leetcode_ContainsDuplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        dictNums = {}
        retVal = False
        
        for element in nums:
            if element not in dictNums:
                dictNums[element] = 1
            else:
                dictNums[element] += 1
                return True
        
        return retVal
#         for key,value in dictNums.items():
#             if value > 1:
#                 retVal = True
        
#         return retVal

File: 217Leetcode_ContainsDuplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        
        map<int,int> mp;
        
        for(int i = 0; i < nums.size(); i++)
        {
            mp[nums[i]] += 1;
        }
        
        for(auto x:mp)
        {
            //cout << x.first << " " << x.second << endl;
            if(x.second > 1)
            {
                return true;
            }
        }
        
        return false;
    }
};